tests/MongoDB.Driver.Tests/Specifications/command-monitoring/CommandMonitoringTestRunner.cs

```diff
@@ -219,9 +234,39 @@ private void ExecuteOperation(IMongoDatabase database, IMongoCollection<BsonDocu
                 throw new SkipException(reason);
             }
 
+            var collection = GetCollection(database, collectionName, operation);
             test.Execute(__client.Cluster.Description, database, collection, arguments, async);
         }
 
+        private void ParseCollectionOptions(MongoCollectionSettings settings, BsonDocument collectionOptions)
+        {
+            foreach (var collectionOption in collectionOptions.Elements)
+            {
+                switch (collectionOption.Name)
+                {
+                    case "writeConcern":
+                        settings.WriteConcern = WriteConcern.FromBsonDocument(collectionOption.Value.AsBsonDocument);
+                        break;
+                    default:
+                        throw new FormatException($"Unexpected collection option: {collectionOption.Name}.");
+                }
+            }
+        }
+
+        private MongoCollectionSettings ParseCollectionOptions(BsonDocument operation)
```

Comment:
Rename this method something like `GetOperationCollectionOptions` so that we don't have a name collision on two methods with the same name and the same signature.

In other words, use different names rather than artificially different signatures to disambiguate the methods.

Reply:
This was not the naming I was suggesting. I propose:

    private MongoCollectionSettings ParseOperationCollectionSettings(BsonDocument operation)
    {
        if (operation.TryGetValue("collectionOptions", out var collectionOptions))
        {
            return ParseCollectionSettings(collectionOptions.AsBsonDocument);
        }
        else
        {
            return null;
        }
    }

Note: I now think `ParseOperationCollectionSettings` is a better name than `GetOperationCollectionOptions`.

`ParseCollectionOptions` implies that the return type is `CollectionOptions` but there is no such type.